src/saveDocs.ts

```diff
@@ -97,9 +103,11 @@ async function addMetaDatas(pkgs: RawPkg[]): Promise<FinalPkg[]> {
     jsDelivr.getAllFilesList(pkgs),
   ]);
 
-  const [changelogs, ts] = await Promise.all([
+  const [changelogs, ts, moduleTypes, styleTypes] = await Promise.all([
     getChangelogs(pkgs, filelists),
     getTSSupport(pkgs, filelists),
+    getModuleTypesForAll(pkgs, filelists),
```

Comment:
this now calculates moduleTypes both during formatPkg and here, won't one always override the other?

Reply:
If `moduleTypes` is resolved to anything different from `unknown` in formatPkg (package.json based), no further calculation will be made. For this field, it's enough to know that any of `esm` or `cjs` is detected.
But if it's `unknown` we must check if there are js files at all.

Reply:
But this will not cover the case when package.json is:

Reply:
invalid packages like that is a complicated situation. I think it's probably okay to consider that an anomaly and leave it as "undefined behaviour". In this case that would be something like marking it as cjs when really we don't know what type it is is probably fine